src/demo-app/checkbox/checkbox-demo.ts

```diff
@@ -0,0 +1,97 @@
+import {Component} from 'angular2/core';
+import {MdCheckbox} from '../../components/checkbox/checkbox';
+
+interface Task {
+  name: string,
+  completed: boolean,
+  subtasks?: Task[]
+}
+
+@Component({
+  selector: 'md-checkbox-demo-nested-checklist',
+  styles: [`
+    li {
+      margin-bottom: 4px;
+    }
+  `],
+  template: `
+    <h2>Tasks</h2>
```

Comment:
Seemed like overkill. I'm from the camp that likes everything in one file (scripts, styles, templates, etc.) barring a _major_ file size overhaul. Keeping everything in one place reduces developer fatigue of jumping around files to edit tightly coupled dependencies, and creates one singular source of truth for component information. Furthermore, barring some exception (e.g. the plethora of animations and style changes material design input controls have to acommodate, or just using preprocessors in general), if a component's DOM / styles get too large, it's probably a good indication that it's time to refactor that component into smaller ones.

WDYT? I'm happy to move it into its own file, IDC that much either way :smiley: